src/components/policy/policy_regular/src/cache_manager.cc

```diff
@@ -1226,6 +1242,46 @@ void CacheManager::CheckSnapshotInitialization() {
   }
 }
 
+policy_table::VehicleDataItems CacheManager::CalculateCustomVdItemsDiff(
+    const policy_table::VehicleDataItems& items_before,
+    const policy_table::VehicleDataItems& items_after) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (items_before.empty()) {
+    LOG4CXX_DEBUG(logger_, "No custom VD items found in policy");
+    return policy_table::VehicleDataItems();
+  }
+
+  if (items_after.empty()) {
+    LOG4CXX_DEBUG(logger_,
+                  "All custom VD items were removed after policy update");
+    return items_before;
+  }
+
+  policy_table::VehicleDataItems removed_items;
+  for (auto& item_to_search : items_before) {
+    auto item_predicate =
+        [&item_to_search](const policy_table::VehicleDataItem& item_to_check) {
+          return item_to_search.name == item_to_check.name;
+        };
+
+    auto it =
+        std::find_if(items_after.begin(), items_after.end(), item_predicate);
+    if (items_after.end() == it) {
+      removed_items.push_back(item_to_search);
+    }
+  }
+
+  LOG4CXX_DEBUG(logger_,
+                "Was found " << removed_items.size() << " removed VD items");
```

Comment:
```suggestion
                "Found " << removed_items.size() << " removed custom VD items");
```

Reply:
@ShobhitAd fixed in https://github.com/smartdevicelink/sdl_core/pull/3081/commits/b6060b86c65aa0d5fedc207fb399ad9fcfbc9e4b